File: src/utils/env.py
import os
from dotenv import load_dotenv
from pydantic import BaseModel, Field
from typing import Optional
from src.banks.types import BankName

load_dotenv()

class Env(BaseModel):
    YNAB_BUDGET_ID: str
    DEBUG: Optional[bool] = Field(default=None)
    TZ: str
    UUID_NAMESPACE: str
    JMAP_SESSION_URL: str
    SECRET_NAME: str
    AWS_ACCESS_KEY_ID: Optional[str] = Field(default=None)
    AWS_SECRET_ACCESS_KEY: Optional[str] = Field(default=None)
    AWS_DEFAULT_REGION: Optional[str] = Field(default=None)
    BANK: BankName
    AWS_S3_BUCKET_NAME: Optional[str] = Field(default=None)

env = Env(
    YNAB_BUDGET_ID=os.getenv("YNAB_BUDGET_ID"),
    DEBUG=os.getenv("DEBUG", "false").lower() in ("true", "1", "t"),
    TZ=os.getenv("TZ"),
    UUID_NAMESPACE=os.getenv("UUID_NAMESPACE"),
    JMAP_SESSION_URL=os.getenv("JMAP_SESSION_URL"),
    SECRET_NAME=os.getenv("SECRET_NAME"),
    AWS_ACCESS_KEY_ID=os.getenv("AWS_ACCESS_KEY_ID"),
    AWS_SECRET_ACCESS_KEY=os.getenv("AWS_SECRET_ACCESS_KEY"),
    AWS_DEFAULT_REGION=os.getenv("AWS_DEFAULT_REGION"),
    BANK=BankName(os.getenv("BANK")),
    AWS_S3_BUCKET_NAME=os.getenv("AWS_S3_BUCKET_NAME"),
)

File: src/utils/env.ts
import { z } from "zod";
import { BankName } from "../banks/types";

const Env = z.object({
  TZ: z.string(),
  DEBUG: z.coerce.boolean().default(false),
  UUID_NAMESPACE: z.string().default("f47ac10b-58cc-4372-a567-0e02b2c3d479"),
  YNAB_BUDGET_ID: z.string().default("last-used"),
  JMAP_SESSION_URL: z.string().default("https://api.fastmail.com/jmap/session"),
  HEADLESS: z.coerce.boolean().default(false),
  AWS_ACCESS_KEY_ID: z.string().optional(),
  AWS_SECRET_ACCESS_KEY: z.string().optional(),
  AWS_DEFAULT_REGION: z.string().optional(),
  AWS_DYNAMODB_MESSAGES_TABLE_NAME: z.string(),
  AWS_DYNAMODB_PENDING_TRANSACTIONS_TABLE_NAME: z.string(),
  TRANSACTIONS_WEBHOOK_URL: z.string(),
  PENDING_NOTIFICATIONS_ENABLED: z.coerce.boolean().default(true),
  HTTP_PROXY: z.string().optional(),
  BANK: z.enum(BankName),
  TRACES_PATH: z.string(),
  USER_DATA_PATH: z.string(),
});

export default Env.parse(Bun.env);
